presto-spi/src/main/java/com/facebook/presto/spi/page/PagesSerdeUtil.java

```diff
@@ -69,8 +86,24 @@ public static SerializedPage readSerializedPage(SliceInput sliceInput)
         byte codecMarker = sliceInput.readByte();
         int uncompressedSizeInBytes = sliceInput.readInt();
         int sizeInBytes = sliceInput.readInt();
+        long checksum = 0;
+        String host = "";
+        if (CHECKSUMMED.isSet(codecMarker)) {
+            checksum = sliceInput.readLong();
+            int length = sliceInput.readInt();
+            byte[] hostAddress = new byte[length];
+            sliceInput.read(hostAddress);
+            host = new String(hostAddress);
+        }
         Slice slice = sliceInput.readSlice(toIntExact((sizeInBytes)));
-        return new SerializedPage(slice, codecMarker, positionCount, uncompressedSizeInBytes);
+        SerializedPage page = new SerializedPage(slice, codecMarker, positionCount, uncompressedSizeInBytes);
+
+        if (CHECKSUMMED.isSet(codecMarker)) {
+            if (checksum != page.getChecksum()) {
```

Comment:
this will not work. You should use .equals(), ==/!= is comparing its its same reference